presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java

```diff
@@ -0,0 +1,170 @@
+import com.facebook.presto.transaction.TransactionManager;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@ThriftService(value = "presto-catalog-server", idlName = "PrestoCatalogServer")
+public class CatalogServer
+{
+    private static final String EMPTY_STRING = "";
+    private final Metadata metadataProvider;
+    private final SessionPropertyManager sessionPropertyManager;
+    private final TransactionManager transactionManager;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public CatalogServer(MetadataManager metadataProvider, SessionPropertyManager sessionPropertyManager, TransactionManager transactionManager, ObjectMapper objectMapper)
+    {
+        this.metadataProvider = requireNonNull(metadataProvider, "metadataProvider is null");
+        this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+        this.objectMapper = requireNonNull(objectMapper, "handleResolver is null");
+    }
+
+    @ThriftMethod
+    public boolean schemaExists(TransactionInfo transactionInfo, SessionRepresentation session, CatalogSchemaName schema)
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        return metadataProvider.schemaExists(session.toSession(sessionPropertyManager), schema);
+    }
+
+    @ThriftMethod
+    public boolean catalogExists(TransactionInfo transactionInfo, SessionRepresentation session, String catalogName)
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        return metadataProvider.catalogExists(session.toSession(sessionPropertyManager), catalogName);
+    }
+
+    @ThriftMethod
+    public String listSchemaNames(TransactionInfo transactionInfo, SessionRepresentation session, String catalogName)
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        List<String> schemaNames = metadataProvider.listSchemaNames(session.toSession(sessionPropertyManager), catalogName);
+        if (!schemaNames.isEmpty()) {
+            return writeValueAsString(schemaNames, objectMapper);
+        }
+        return EMPTY_STRING;
+    }
+
+    @ThriftMethod
+    public String getTableHandle(TransactionInfo transactionInfo, SessionRepresentation session, QualifiedObjectName table)
+    {
+        transactionManager.tryRegisterTransaction(transactionInfo);
+        Optional<TableHandle> tableHandle = metadataProvider.getTableHandle(session.toSession(sessionPropertyManager), table);
+        if (tableHandle.isPresent()) {
+            return writeValueAsString(tableHandle.get(), objectMapper);
+        }
+        return EMPTY_STRING;
```

Comment:
replace all the `isPresent` checks with functional style  `map`
e.g.
```
        return tableHandle.map(handle -> writeValueAsString(handle, objectMapper))
                .orElse(EMPTY_STRING);
```

Reply:
Added to all isPresent checks

Reply:
Added to all isPresent checks